src/nft/components/common/SortDropdown/SortDropdown.tsx

```diff
@@ -0,0 +1,239 @@
+import clsx from 'clsx'
+import { useOnClickOutside } from 'hooks/useOnClickOutside'
+import { Box } from 'nft/components/Box'
+import { Row } from 'nft/components/Flex'
+import { ArrowsIcon, ChevronUpIcon, ReversedArrowsIcon } from 'nft/components/icons'
+import { buttonTextMedium } from 'nft/css/common.css'
+import { themeVars } from 'nft/css/sprinkles.css'
+import { DropDownOption } from 'nft/types'
+import { useEffect, useLayoutEffect, useMemo, useReducer, useRef, useState } from 'react'
+
+import * as styles from './SortDropdown.css'
+
+export const SortDropdown = ({
+  dropDownOptions,
+  inFilters,
+  mini,
+  miniPrompt,
+  top,
+  left,
+}: {
+  dropDownOptions: DropDownOption[]
+  inFilters?: boolean
+  mini?: boolean
+  miniPrompt?: string
+  top?: number
+  left?: number
+}) => {
+  const [isOpen, toggleOpen] = useReducer((s) => !s, false)
+  const [isReversed, toggleReversed] = useReducer((s) => !s, false)
+  const [selectedIndex, setSelectedIndex] = useState(0)
+
+  const [maxWidth, setMaxWidth] = useState(0)
+
+  const ref = useRef<HTMLDivElement>(null)
+  useOnClickOutside(ref, () => isOpen && toggleOpen())
+
+  useEffect(() => setMaxWidth(0), [dropDownOptions])
+
+  const reversable = useMemo(
+    () => dropDownOptions[selectedIndex].reverseOnClick || dropDownOptions[selectedIndex].reverseIndex,
+    [selectedIndex, dropDownOptions]
+  )
+
+  return (
+    <Box
+      ref={ref}
+      transition="250"
+      borderRadius="12"
+      borderBottomLeftRadius={isOpen ? '0' : undefined}
```

Comment:
this is declared a few times, can this be a constant?

Reply:
What are you thinking can become a constant? We have a couple of different styles that are conditional based on the isOpen param which is why it is used in several spots

Reply:
```isOpen ? '0' : undefined```

this styling is used 4 times